clap_complete/src/dynamic/shells/fish.rs

```diff
@@ -30,8 +30,12 @@ impl crate::dynamic::Completer for Fish {
         let index = args.len() - 1;
         let completions = crate::dynamic::complete(cmd, args, index, current_dir)?;
 
-        for completion in completions {
-            writeln!(buf, "{}", completion.to_string_lossy())?;
+        for (completion, help) in completions {
+            write!(buf, "{}", completion.to_string_lossy())?;
+            if let Some(help) = help {
+                write!(buf, "\t{help}")?;
```

Comment:
Do we need to escape anything in `help`?

Reply:
Well it probably shouldn't be multiline, at some point it would maybe be great to be able to have a custom description for completions.

For now, we could either, just take the first line or replace new lines with spaces.

Reply:
As a reminder to both of us, `Display` for `StyledStr` will strip out ANSI escape codes, so we don't need to worry about any policies related to that with how this is written.  So we could do `help.to_string()` and split off the first line.

Any concern over tabs, [quotes, backslashes, or commas](https://github.com/clap-rs/clap/blob/master/clap_complete/src/shells/fish.rs)

Reply:
> Any concern over tabs, [quotes, backslashes, or commas](https://github.com/clap-rs/clap/blob/master/clap_complete/src/shells/fish.rs)

I don't think so, but I'll create a test to make sure.

Reply:
Don't we at least need to handle the multi-line case?